sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/component.py

```diff
@@ -423,15 +425,21 @@ def _from_rest_object_to_init_params(cls, obj: ComponentVersion) -> Dict:
         # note that there is an issue that environment == {} will always be true, so use isinstance here
         return {k: v for k, v in init_kwargs.items() if v is not None and not (isinstance(v, dict) and not v)}
 
-    def _get_anonymous_hash(self) -> str:
+    def _get_anonymous_hash(self, keys_to_omit=None) -> str:
         """Return the name of anonymous component.
 
         same anonymous component(same code and interface) will have same name.
         """
         component_interface_dict = self._to_dict()
         # omit version since anonymous component's version is random guid
         # omit name since name doesn't impact component's uniqueness
-        return hash_dict(component_interface_dict, keys_to_omit=["name", "id", "version"])
+        return hash_dict(
+            component_interface_dict, keys_to_omit=["name", "id", "version"] if keys_to_omit is None else keys_to_omit
+        )
+
+    def get_component_hash(self, keys_to_omit=None) -> str:
```

Comment:
This will change public API. Why we should expose this to users?

Reply:
It not need, I will update to private.